code/modules/power/tesla/energy_ball.dm

```diff
@@ -109,13 +112,13 @@
 		switch(z_move)
 			if(1)
 				visible_message(span("danger","\The [src] gravitates upwards!"))
-				forceMove(T)
+				zMove(UP)
 				visible_message(span("danger","\The [src] gravitates from below!"))
 			if(0)
-				forceMove(T)
+				walk_to(src, T, 0, speed)
```

Comment:
psure there's an issue with `walk_to` that stops clean destroy. Do you have to use it? If you do, then make sure to cancel the `walk_to` in `energy_ball/Destroy()`. Unless it's already done at the atom level. 🤔

Reply:
I do want to use it. I will add cancellation to `Destroy()`